e2e: Re-enable and fix flaky pipelinerun-with-matrix example test

## Problem

  The `TestExamples/v1/pipelineruns/beta/pipelinerun-with-matrix` e2e test has been consistently flaky in CI, causing build failures. The test file has been temporarily moved to the `no-ci` folder to prevent CI failures while we investigate and fix
  the root cause.

  The test was disabled in PR #9200 as a temporary measure.

  ## Evidence of Flakiness

  From issue #9062, the test has been failing with timeouts:

  --- FAIL: TestExamples/v1/pipelineruns/beta/pipelinerun-with-matrix (605.80s)

  Recent CI failures:
  - https://github.com/tektoncd/pipeline/actions/runs/18324465586/job/52185466094?pr=9047
  - https://github.com/tektoncd/pipeline/actions/runs/18324465586/job/52185466098?pr=9047
  - https://github.com/tektoncd/pipeline/actions/runs/18339183125/job/52230530489?pr=9067

  ## Tasks

  - [ ] Investigate why the pipelinerun-with-matrix example test is timing out (600+ seconds)
  - [ ] Identify the root cause of the flakiness
  - [ ] Implement a fix for the underlying issue
  - [ ] Move the test file back from `examples/v1/pipelineruns/no-ci/pipelinerun-with-matrix.yaml` to `examples/v1/pipelineruns/beta/pipelinerun-with-matrix.yaml`
  - [ ] Verify the test runs reliably in CI

  ## Current Location

  The test file is currently at: `examples/v1/pipelineruns/no-ci/pipelinerun-with-matrix.yaml`

  It should be moved back to: `examples/v1/pipelineruns/beta/pipelinerun-with-matrix.yaml` once fixed.

  ## Related Issues

  - #9062 - Parent issue tracking flaky e2e example tests

--- a/examples/v1/pipelineruns/beta/pipelinerun-with-matrix.yaml
+++ b/examples/v1/pipelineruns/beta/pipelinerun-with-matrix.yaml
@@ -32,7 +32,6 @@ spec:
             - name: platform
               value:
                 - linux
-                - mac
             - name: browser
               value:
                 - chrome
@@ -45,7 +44,6 @@ spec:
             - name: platform
               value:
                 - linux
-                - mac
         params:
           - name: browser
             value: chrome
@@ -81,7 +79,6 @@ spec:
             - name: version
               value:
                 - "1"
-                - "2"
         taskSpec:
           params:
             - name: version
@@ -100,7 +97,6 @@ spec:
             - name: version
               value:
                 - "1"
-                - "2"
         taskSpec:
           params:
             - name: version

--- a/examples/v1/pipelineruns/beta/using-retries-and-retry-count-variables.yaml
+++ b/examples/v1/pipelineruns/beta/using-retries-and-retry-count-variables.yaml
@@ -8,21 +8,18 @@ spec:
   pipelineSpec:
     tasks:
     - name: retry-me
-      retries: 5
+      retries: 1
       params:
       - name: pipelineTask-retries
         value: "$(context.pipelineTask.retries)"
-      - name: pipelineTask-retry-count
-        value: "$(context.task.retry-count)"
       taskSpec:
         params:
         - name: pipelineTask-retries
-        - name: pipelineTask-retry-count
         steps:
         - image: mirror.gcr.io/alpine
           script: |
             #!/usr/bin/env sh
-            if [ "$(params.pipelineTask-retry-count)" == "$(params.pipelineTask-retries)" ]; then
+            if [ "$(context.task.retry-count)" == "$(params.pipelineTask-retries)" ]; then
               echo "This is the last retry."
               exit 0;
             fi

